game/src/Game/LoadingScreen.gd

```diff
@@ -0,0 +1,30 @@
+extends Control
+class_name LoadingScreen
+
+@export var progress_bar: ProgressBar
+@export var quote_label: Label
+
+var loadthread: Thread
+var quotes: PackedStringArray = []
+
+func update_loading_screen(percent_complete: int, quote_should_change = false):
+	progress_bar.value = percent_complete
+	if quote_should_change:
+		quote_label.text = quotes[randi() % quotes.size()]
+
+func _on_splash_container_splash_end():
+	show()
+
+func _ready():
+	# FS-3, UI-30, UIFUN-35
+	loadthread = Thread.new()
+	loadthread.start(Events.load_events.bind(self))
```

Comment:
Probably best to consider [WorkerThreadPool](https://docs.godotengine.org/en/latest/classes/class_workerthreadpool.html) (yes this documentation is 4.1+, but the API should not change, the backend is better in 4.1, but how you use it should not change regardless) whether now or later.